File: 0169-majority-element.cpp
/*
169. Majority Element

Submitted: December 5, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 28.15 MB (beats 20.01%)
*/

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> map;
        for ( const int num : nums ) {
            map[num]++;
        }
        for ( auto& kv : map ) {
            // problem statement says that floor(n / 2) is majority
            if (kv.second > (nums.size() / 2)) return kv.first;
        }
        return -1;
    }
};

File: 0169-majority-element.py
"""
169. Majority Element

Submitted: December 14, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 19.25 MB (beats 19.25%)
"""

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        return nums[len(nums) // 2]